log/splunk/splunk.go

```diff
@@ -0,0 +1,141 @@
+package splunk
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"time"
+)
+
+type Event struct {
+	Time       int64       `json:"time"`                 // epoch time in seconds
+	Host       string      `json:"host"`                 // hostname
+	Source     string      `json:"source,omitempty"`     // optional description of the source of the event; typically the app's name
+	SourceType string      `json:"sourcetype,omitempty"` // optional name of a Splunk parsing configuration; this is usually inferred by Splunk
+	Index      string      `json:"index,omitempty"`      // optional name of the Splunk index to store the event in; not required if the token has a default index set in Splunk
+	Event      interface{} `json:"event"`                // throw any useful key/val pairs here
+}
+
+type Client struct {
+	HTTPClient *http.Client
+	URL        string
+	Hostname   string
+	Token      string
+	Source     string
+	SourceType string
+	Index      string
+}
+
+func NewClient(httpClient *http.Client, URL string, Token string, Source string, SourceType string, Index string) *Client {
```

Comment:
This constructor is a bit annoying because it requires all those positional parameters. It would be nicer to allow creating the client as a literal, filling in the struct fields you want and leaving the ones you don't need as zero.

```go
client := &splunk.Client{
    ...
}
client.Log(...)
```

To make this work, add a lazy initializer which will fill in the HTTP client unless it is already set.

Reply:
OK, I think I've gone and done that.